### PR TITLE
Basic support for definition export

### PR DESCRIPTION
**Pull request by pathcl**
Hello,

Nice library! thanks for making it open source. I tried to find a way to get whole rabbitmq configuration but couldn't find it.

Idea here is to get full config from rabbitmq runtime. Eventually this could be used along with terraform to eventually import/manage it.

### CODE REVIEW

**Review by michaelklishin — CHANGES_REQUESTED** [review]
Thank you for taking the time to contribute.

I'm afraid using interfaces for most entity collections is not something we would be interested in doing. This significantly reduces the amount of effort involved but makes it very hard to reason about the structure of responses.

You can use [Hop](https://github.com/rabbitmq/hop) as an example of a library that supports definition import and export in a more type-safe manner.

I'd understand if you decide against spending more time on this PR.

**Comment by pathcl** [comment]
Hey,

Would it be ok if I replace all interfaces for specific types? I can
improve it

**Comment by michaelklishin** [comment]
@pathcl sure, that's what I had in mind. Thank you. Can we please introduce a new type for global parameters? I can help get it over the finish line do it if you are not familiar with them.

**Review by michaelklishin — CHANGES_REQUESTED** [review]
This is definitely a step in the right direction. Let's also cover global parameters.

File: runtime_parameters.go [hunk 1]
```diff
@@ -16,9 +16,44 @@ type RuntimeParameter struct {
 	Value     interface{} `json:"value"`
 }
 
+// RuntimeDefinitions represents rabbitmq runtime configuration.
+type RuntimeDefinitions struct {
```
Comment (michaelklishin) [inline runtime_parameters.go:20]: They are called just [definitions](https://www.rabbitmq.com/definitions.html), not runtime definitions. A lot of them have nothing to do with runtime parameters and do not change at runtime (e.g. product version).

File: runtime_parameters.go [hunk 2]
```diff
@@ -16,9 +16,44 @@ type RuntimeParameter struct {
...
+}
+
 // RuntimeParameterValue represents arbitrary parameter data.
 type RuntimeParameterValue map[string]interface{}
 
+//
+// GET /api/definitions
+//
+
+func (c *Client) ListRuntimeDefinitions() (p *RuntimeDefinitions, err error) {
```
Comment (michaelklishin) [inline runtime_parameters.go:44]: This should be renamed per the comment above.

**Comment by michaelklishin** [comment]
Finished this and merged manually. Thank you. We may want to compare the global runtime parameters serialization to that of Hop but I'm reasonably confident in this feature as is.

**Pull request merged**